pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -4305,6 +4356,316 @@ def PumpCostRule(b, l1, l2):
             doc="Objective function",
         )
 
+    elif model.config.hydraulics == Hydraulics.co_optimize_linearized:
+        """
+        For the co-optimize method, the hydraulics block is solved along with
+        the network model to optimize the network in the presence of
+        hydraulics constraints. The objective is to minimize total cost including
+        the cost of pumps.
+        """
+        # In the co_optimize method the objective should include the cost of pumps. So,
+        # delete the original objective to add a modified objective in this method.
+        del model.objective
+
+        # add necessary variables and constraints
+        mh.v_HW_loss = Var(
+            model.s_LLA,
+            model.s_T,
+            initialize=0,
+            within=NonNegativeReals,
+            units=pyunits.meter,
+            doc="Hazen-Williams Frictional loss, m",
+        )
+
+        mh.v_variable_pump_cost = Var(
+            model.s_LLA,
+            model.s_T,
+            initialize=0,
+            within=NonNegativeReals,
+        )
+
+        # -----------------Removing effective diameter rule. We will put the expression of binaries where needed
+        # mh.v_effective_Pipeline_diameter = Var(
```

Comment:
Is there a reason for keeping this block of commented code in the file? In general, unless there's a reason, we want to avoid having blocks of commented out code like this.

Reply:
Sure, I have removed the comments.